src/app/parcels/form/clientDatabaseFunctions.ts

```diff
@@ -2,34 +2,71 @@ import supabase from "@/supabaseClient";
 import { InsertSchema, UpdateSchema } from "@/databaseUtils";
 import { DatabaseError } from "@/app/errorClasses";
 import { logErrorReturnLogId } from "@/logger/logger";
+import { AuditLogProps, sendAuditLog } from "@/server/auditLog";
 
 type ParcelDatabaseInsertRecord = InsertSchema["parcels"];
 type ParcelDatabaseUpdateRecord = UpdateSchema["parcels"];
 
 export const insertParcel = async (parcelRecord: ParcelDatabaseInsertRecord): Promise<void> => {
-    const { error } = await supabase
+    const { data, error } = await supabase
         .from("parcels")
         .insert(parcelRecord)
-        .select("primary_key, client_id");
+        .select("primary_key, client_id")
+        .single();
+
+    const auditLog: AuditLogProps = {
+        action: "add a parcel",
+        content: { parcelDetails: parcelRecord },
+        wasSuccess: false,
+        clientId: parcelRecord.client_id,
+        collectionCentreId: parcelRecord.collection_centre
+            ? parcelRecord.collection_centre
+            : undefined,
+        packingSlotId: parcelRecord.packing_slot ? parcelRecord.packing_slot : undefined,
+    };
 
     if (error) {
         const logId = await logErrorReturnLogId("Error with insert: parcel data", error);
+        auditLog.logId = logId;
+        await sendAuditLog(auditLog);
         throw new DatabaseError("insert", "parcel data", logId);
     }
+
+    auditLog.parcelId = data.primary_key;
+    auditLog.wasSuccess = true;
+    await sendAuditLog(auditLog);
```

Comment:
I'm a little wary of changing an object for readability, so it might be better to use `...` to construct a new object to pass to `sendAuditLog`. (similarly in the updateParcel function)

```suggestion
    const auditLog = {
        action: "add a parcel",
        content: { parcelDetails: parcelRecord },
        clientId: parcelRecord.client_id,
        collectionCentreId: parcelRecord.collection_centre
            ? parcelRecord.collection_centre
            : undefined,
        packingSlotId: parcelRecord.packing_slot ? parcelRecord.packing_slot : undefined,
    } as const satisfies Partial<AuditLogProps>;

    if (error) {
        const logId = await logErrorReturnLogId("Error with insert: parcel data", error);
        await sendAuditLog({ ...auditLog, wasSuccess: false, logId });
        throw new DatabaseError("insert", "parcel data", logId);
    }

    await sendAuditLog({ ...auditLog, wasSuccess: true, parcelId: data.primary_key });
```